mempool/src/shared_mempool/coordinator.rs

```diff
@@ -67,6 +68,17 @@ pub(crate) async fn coordinator<V>(
     let workers_available = smp.config.shared_mempool_max_concurrent_inbound_syncs;
     let bounded_executor = BoundedExecutor::new(workers_available, executor.clone());
 
+    let initial_reconfig = mempool_reconfig_events
```

Comment:
Previously mempool was not reading the initial config before starting, which seems like an oversight.

This change required a lot of refactoring of the tests. `MockDbReaderWriter` could not handle the on chain configs.